app/lib/surface/Label.coffee

```diff
@@ -46,17 +47,27 @@ module.exports = class Label extends CocoClass
     o = @buildLabelOptions()
     @layer.addChild @label = @buildLabel o
     @layer.addChild @background = @buildBackground o
+    @returnBounds = o.returnBounds
     @layer.updateLayerOrder()
 
   update: ->
     return unless @text and @sprite.sprite
     offset = @sprite.getOffset? (if @style in ['dialogue', 'say'] then 'mouth' else 'aboveHead')
-    offset ?= x: 0, y: 0  # temp (if not Lank)
+    offset ?= { x: 0, y: 0 }  # temp (if not Lank)
     offset.y += 10 if @style is 'variable'
     rotation = @sprite.getRotation()
     offset.x *= -1 if rotation >= 135 or rotation <= -135
     @label.x = @background.x = @sprite.sprite.x + offset.x
     @label.y = @background.y = @sprite.sprite.y + offset.y
+    if @returnBounds and @background.bitmapCache and camera = @sprite?.options?.camera
```

Comment:
This assignment in an `if` statement looks fine here, but in general I find these highly confusing. My mind expects to find only conditionals in the check, and when there's an assignment-turn-conditional I immediately wonder if there is a missing `=`, do a double take, and then read through again to check the usage. 

It's less efficient, and +1 LOC, to move the assignment above and then do the check. It's +2 LOC to move the check inside the if and add another if. Still, worth it to not do the double take on the assignment-turn-conditional in my personal opinion. 

Thoughts on assignments in `if`'s in general @codecombat/core ?

Reply:
I don't like assignments in if statements (I don't like it anywhere except newlines) because I'm Python developer first :-)
Here I just copied the code style from that file, I thought it's some convention maybe. In component code, I'm trying to get rid of that.

As for this line, I think the whole file should be changed at all, because there are more statements like this. How about put it when we rework it in JS?